ConsoleFramework/Controls/Window.cs

```diff
@@ -16,6 +16,7 @@ public class Window : Control
         public static RoutedEvent ActivatedEvent = EventManager.RegisterRoutedEvent("Activated", RoutingStrategy.Direct, typeof(EventHandler), typeof(Window));
         public static RoutedEvent DeactivatedEvent = EventManager.RegisterRoutedEvent("Deactivated", RoutingStrategy.Direct, typeof(EventHandler), typeof(Window));
         public static RoutedEvent ClosedEvent = EventManager.RegisterRoutedEvent("Closed", RoutingStrategy.Direct, typeof(EventHandler), typeof(Window));
+        public static RoutedEvent ClosingEvent = EventManager.RegisterRoutedEvent("Closing", RoutingStrategy.Direct, typeof(RoutedEventHandler), typeof(Window));

```

Comment:
I suggest to create a new class

`public delegate void CancelEventHandler(object sender, CancelEventArgs e);`

and `CancelEventArgs : RoutedEventArgs` with `bool Cancel` property.

And instead of checking args.Handled check args.Cancel in `handleClosing()`.

Just like it is made in WPF :)

https://docs.microsoft.com/en-us/dotnet/api/system.componentmodel.canceleventhandler?view=netframework-4.8
https://docs.microsoft.com/en-us/dotnet/api/system.componentmodel.canceleventargs?view=netframework-4.8

Reply:
Oh yes, i had even done that, but I thought I was having a problem with my code and went back in the code, I will add it again.

Reply:
Done!

Reply:
:+1: